src/pages/index.js

```diff
@@ -1,67 +1,214 @@
+    renderer: (cardData) => {
+      const card = new Card(
+        cardData,
+        cardTemplate,
+        (cardData) => imagePopup.open(cardData),
+        handleTrashClick,
+        api
+      ).getView();
+      if (cardData.isLiked) {
+        card
+          .querySelector(".card__like-button")
+          .classList.add("card__like-button_active");
+      }
+      cardSection.appendItem(card);
+    },
+  },
+  ".cards__list"
+);
+
+// -------------------------------------------------------------------------------------
+// Initial Card Creation & Rendering from API
+// -------------------------------------------------------------------------------------
+api
+  .getCardList()
+  .then((cardList) => {
+    cardSection.renderItems(cardList);
+  })
+  .catch((error) => {
+    console.error("Failed to fetch or render cards:", error);
+  });
+
+// -------------------------------------------------------------------------------------
+// Additional Card adding to API & DOM
+// -------------------------------------------------------------------------------------
 const addCardPopup = new PopupWithForm("#add-card-modal", {
   handleFormSubmit: (formData) => {
     const cardData = {
       name: formData["title"],
       link: formData["Image URL"],
     };
-    const card = createCard(cardData);
-    cardSection.prependItem(card);
-    addCardPopup.getForm().reset();
-    formValidators[
-      addCardPopup.getForm().getAttribute("name")
-    ].toggleButtonState();
-    addCardPopup.close();
+
+    addCardPopup.toggleUploadIndicator(true);
+
+    api
+      .addCard(cardData)
+      .then((apiCardData) => {
+        const card = new Card(
+          apiCardData,
+          cardTemplate,
+          (data) => imagePopup.open(data),
+          handleTrashClick,
+          api
+        ).getView();
+        cardSection.prependItem(card);
+        const validator = formValidators["add-card-form"];
+        validator.resetValidation();
+        addCardPopup.close();
+      })
+      .catch((error) => console.error("Error adding card:", error))
+      .finally(() => addCardPopup.toggleUploadIndicator(false));
   },
 });
+
 addCardPopup.setEventListeners();
 
-const imagePopup = new PopupWithImage("#image-modal");
-imagePopup.setEventListeners();
 // -------------------------------------------------------------------------------------
-//                               Validation
+// Card deletion from API & DOM
+// -------------------------------------------------------------------------------------
+const deleteCardModal = new PopupWithConfirm("#delete-card-modal", () => {});
+deleteCardModal.setEventListeners();
+
+function handleTrashClick(card) {
+  deleteCardModal.open();
+  deleteCardModal.setSubmitHandler(() => {
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        card.element.remove();
+        card.element = null;
+      })
+      .catch((error) => console.error("Failed to delete card:", error));
+    deleteCardModal.close();
```

Comment:
* You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**